pkg/v1/config/clientconfig.go

```diff
@@ -507,3 +580,30 @@ func IsFeatureActivated(feature string) bool {
 	}
 	return status
 }
+
+// GetDiscoverySources returns all discovery sources
+// Includes standalone discovery sources and if server is available
+// it also includes context based discovery sources as well
+func GetDiscoverySources(serverName string) []configv1alpha1.PluginDiscovery {
+	server, err := GetServer(serverName)
+	if err != nil {
+		log.Warningf("unknown server '%s', Unable to get server based discovery sources: %s", serverName, err.Error())
+		return []configv1alpha1.PluginDiscovery{}
+	}
+
+	discoverySources := server.DiscoverySources
+	// If current server type is management-cluster, then add
+	// the default kubernetes discovery endpoint pointing to the
+	// management-cluster kubeconfig
+	if server.Type == configv1alpha1.ManagementClusterServerType {
+		defaultClusterK8sDiscovery := configv1alpha1.PluginDiscovery{
+			Kubernetes: &configv1alpha1.KubernetesDiscovery{
+				Name:    fmt.Sprintf("default-%s", serverName),
+				Path:    server.ManagementClusterOpts.Path,
+				Context: server.ManagementClusterOpts.Context,
+			},
+		}
+		discoverySources = append(discoverySources, defaultClusterK8sDiscovery)
+	}
```

Comment:
The `if` body doesn't seem to be covered by tests.

This method does get called (indirectly) from manager_test.go, but it doesn't have "its own" unit tests.

Reply:
added unit test for this function which includes `if` body.